apps/hash-graph/lib/graph/src/store/postgres.rs

```diff
@@ -49,164 +37,8 @@ use crate::{
-
-                    TraversalStatus::Unresolved(
-                        *current_depths,
-                        difference
-                            .into_iter()
-                            .next()
-                            .expect("difference must be a single interval"),
-                    )
-                } else {
-                    // The time intervals are disjoint and not adjacent. Ideally, we only would
-                    // resolve the new interval, but we did not come up with a good way to store the
-                    // different intervals in the traversal map. So we resolve the full interval
-                    // for now.
-                    //
-                    // We only require this logic when traversing edges of the graph, so the roots
-                    // of the graph are always precisely resolved correctly. By this implementation,
-                    // we may get more dependencies resolved than necessary, which will appear as
-                    // vertices/edges in the subgraph. until we have decided, if we consider this a
-                    // bug or not, we keep the current behavior
-                    // see https://app.asana.com/0/0/1203774687353264/f.
-                    //
-                    // Ideally, we want to track this occurrence in production, but utilizing our
-                    // current logging strategy does not work well as we would have to log with a
-                    // high severity. see https://app.asana.com/0/0/1203774687353266/f
-                    //
-                    // However, we only have to resolve the difference of the merge of the two, as
-                    // the old interval is already resolved. and the intervals are disjoint:
-                    // Examples |      1      |      B
-                    // =========|=============|=============
-                    // old      | [---)       |       [---)
-                    // new      |       [---) | [---)
-                    // ---------|-------------|-------------
-                    // optimal  |       [---) | [---)
-                    // current  |     [-----) | [-----)
-                    let difference = old_interval.merge(new_interval).difference(old_interval);
-                    debug_assert_eq!(difference.len(), 1, "difference must be a single interval");
-
-                    TraversalStatus::Unresolved(
-                        *current_depths,
-                        difference
-                            .into_iter()
-                            .next()
-                            .expect("difference must be a single interval"),
-                    )
-                }
-            }
-        }
-    }
-}
-
 #[derive(Default)]
-pub struct TraversalContext {
-    pub ontology_traversal_map: TraversalMap<OntologyTypeVertexId>,
-    pub knowledge_traversal_map: TraversalMap<EntityVertexId>,
-}
+pub struct TraversalContext;
```

Comment:
For future knowledge, we're leaving this here to make sure we're accounting for lifetimes, mutability, etc. in the methods that use it (as we will be bringing it back in a new form after this set of PRs)